ui/src/shared/components/ErrorBoundary.tsx

```diff
@@ -24,6 +27,29 @@ class ErrorBoundary extends Component<ErrorBoundaryProps, ErrorBoundaryState> {
     return {error}
   }
 
+  public componentDidCatch(error: Error, errorInfo: ErrorInfo) {
+    // Parse React's info message to provide the component name this error
+    // occured in.
+    //
+    // For example, given the following info message:
+    //
+    //     The above error occurred in the <MePage> component:
+    //         in MePage (created by ErrorBoundary(MePage))
+    //         in ErrorBoundary (created by ErrorBoundary(MePage))
+    //         in ErrorBoundary(MePage) (created by Connect(ErrorBoundary(MePage)))
+    //         in Connect(ErrorBoundary(MePage)) (created by RouterContext)
+    //         in SpinnerContainer (created by SetOrg)
+    //         ...
+    //
+    // We will extract "MePage" as the component name.
+    const componentName = errorInfo.componentStack
+      .trim()
+      .split('\n')
+      .map(s => s.split(' ')[1])[0]
```

Comment:
This itself looks like a runtime error waiting to happen ahah.

Reply:
Hah definitely. As long as `componentStack` is a string, it should be OK though.